ssl/ssl_sess.c

```diff
@@ -59,17 +59,37 @@ __owur static int timeoutcmp(SSL_SESSION *a, SSL_SESSION *b)
  */
 void ssl_session_calculate_timeout(SSL_SESSION *ss)
```

Comment:
Ensure that the function is tested with edge cases, such as very large timeout values and times close to the maximum value of time_t.

Reply:
I would appreciate it if new unit or component tests could be added for this.

Reply:
I can add such test. my question is if such test should be part of this PR or if I should rather file a new PR.
the thing is that this PR is specific to 3.1 and earlier. 3.2 and later use a `safemath` which detects overflow in advance.  I think we might want to add such test case to all branches (eventually).

thanks.

Reply:
I'm okay addressing it in a separate PR. Please ensure you create a new issue for it first so that it would not be missed. Thanks!